src/main/java/seedu/address/model/GenericList.java

```diff
@@ -0,0 +1,50 @@
+package seedu.address.model;
+
+import java.util.List;
+
+/**
+ * A generic list interface for multi-valued {@code Person} attributes.
+ * @param <T> The type of attribute stored in the list.
+ */
+public interface GenericList<T> {
+    /**
+     * Adds an element to the {@code GenericList<T>}.
+     *
+     * @param t The element to be added.
+     * @return The updated {@code GenericList<T>}.
+     */
+    GenericList<T> add(T t);
+
+    /**
+     * Deletes an element from the {@code GenericList<T>}.
+     *
+     * @param index The index of the element to be deleted.
+     * @return The updated {@code GenericList<T>}.
+     */
+    GenericList<T> delete(int index);
+
+    /**
+     * Edits an element in the {@code GenericList<T>}.
+     *
+     * @param index The index of the element to be edited.
+     * @return The updated {@code GenericList<T>}.
+     */
+    GenericList<T> edit(int index, T t);
+
+    T get(int index);
```

Comment:
Should the index be of type Index instead? then the implementation can decide to use zeroBased or oneBased. Not sure if this is the right way but they are using index class to convey this sort of info right (so the places using these methods does not need to care about computing the index)

Reply:
I think the conversion of `Index` to the approriate `int` is already done in the corresponding `Command` classes.

Reply:
Okay sure then LGTM